File: generateIcons.mjs
#!/usr/bin/env node
/**
 * Generate Chrome extension icons from a base 128x128 image.
 * Usage: node generateIcons.mjs path/to/icon128.png
 */

import sharp from 'sharp';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';
import { existsSync, mkdirSync } from 'fs';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

async function generateIcons(baseImagePath) {
  if (!existsSync(baseImagePath)) {
    console.error(`Error: ${baseImagePath} not found`);
    process.exit(1);
  }

  // Create icons directory if it doesn't exist
  const iconsDir = join(__dirname, 'public', 'icons');
  if (!existsSync(iconsDir)) {
    mkdirSync(iconsDir, { recursive: true });
  }

  // Generate icons in different sizes
  const sizes = [16, 48, 128];
  
  console.log(`Generating icons from ${baseImagePath}...`);
  
  // Check if input is already a 128px icon
  const is128pxInput = baseImagePath.includes('icon128.png');
  const sizesToGenerate = is128pxInput ? [16, 48] : sizes;
  
  for (const size of sizesToGenerate) {
    const outputPath = join(iconsDir, `icon${size}.png`);
    
    try {
      await sharp(baseImagePath)
        .resize(size, size, {
          fit: 'contain',
          background: { r: 0, g: 0, b: 0, alpha: 0 }
        })
        .png()
        .toFile(outputPath);
      
      console.log(`✓ Generated ${outputPath} (${size}x${size})`);
    } catch (error) {
      console.error(`Error generating ${outputPath}:`, error.message);
    }
  }
  
  console.log('\nDone! Icons generated successfully.');
}

// Get command line argument
const baseImagePath = process.argv[2];

if (!baseImagePath) {
  console.log('Usage: node generateIcons.mjs path/to/icon128.png');
  console.log('\nExample: node generateIcons.mjs public/icons/icon128.png');
  process.exit(1);
}

generateIcons(baseImagePath).catch(error => {
  console.error('Error:', error);
  process.exit(1);
});

File: generateIcons.py
#!/usr/bin/env python3
"""
Generate Chrome extension icons from a base 128x128 image.
Usage: python3 generateIcons.py path/to/icon128.png
"""

import sys
from PIL import Image
import os

def generate_icons(base_image_path):
    """Generate 16, 48, and 128px icons from a base 128x128 image."""
    if not os.path.exists(base_image_path):
        print(f"Error: {base_image_path} not found")
        return
    
    # Create icons directory if it doesn't exist
    icons_dir = "public/icons"
    os.makedirs(icons_dir, exist_ok=True)
    
    # Open the base image
    base_image = Image.open(base_image_path)
    
    # Generate icons in different sizes
    sizes = [16, 48, 128]
    for size in sizes:
        # Resize the image
        resized = base_image.resize((size, size), Image.Resampling.LANCZOS)
        
        # Save the icon
        output_path = os.path.join(icons_dir, f"icon{size}.png")
        resized.save(output_path, "PNG")
        print(f"Generated {output_path} ({size}x{size})")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python3 generateIcons.py path/to/icon128.png")
        print("\nFor now, creating placeholder icons...")
        # Create placeholder icons using PIL
        from PIL import Image, ImageDraw
        
        icons_dir = "public/icons"
        os.makedirs(icons_dir, exist_ok=True)
        
        for size in [16, 48, 128]:
            img = Image.new('RGB', (size, size), color='#3b82f6')
            draw = ImageDraw.Draw(img)
            # Draw a simple hand wave icon (circle with line)
            center = size // 2
            radius = size // 3
            draw.ellipse([center - radius, center - radius, center + radius, center + radius], 
                        fill='white', outline='white', width=2)
            output_path = os.path.join(icons_dir, f"icon{size}.png")
            img.save(output_path, "PNG")
            print(f"Generated placeholder {output_path} ({size}x{size})")
    else:
        generate_icons(sys.argv[1])

